src/Sylius/Bundle/CoreBundle/CatalogPromotion/CommandHandler/RemoveInactiveCatalogPromotionHandler.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\CatalogPromotion\CommandHandler;
+
+use Doctrine\ORM\EntityManager;
+use SM\Factory\FactoryInterface;
+use SM\StateMachine\StateMachine;
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Command\RemoveInactiveCatalogPromotion;
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Exception\InvalidCatalogPromotionStateException;
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Sylius\Component\Promotion\Model\CatalogPromotionStates;
+use Sylius\Component\Promotion\Model\CatalogPromotionTransitions;
+use Sylius\Component\Promotion\Repository\CatalogPromotionRepositoryInterface;
+
+final class RemoveInactiveCatalogPromotionHandler
+{
+    public function __construct(
+        private CatalogPromotionRepositoryInterface $catalogPromotionRepository,
+        private EntityManager $entityManager,
+        private FactoryInterface $stateMachineFactory,
+    ) {
+    }
+
+    public function __invoke(RemoveInactiveCatalogPromotion $command): void
+    {
+        /** @var CatalogPromotionInterface|null $catalogPromotion */
+        $catalogPromotion = $this->catalogPromotionRepository->findOneBy(['code' => $command->code]);
+
+        if (null === $catalogPromotion) {
+            return;
+        }
+
+        /** @var StateMachine $stateMachine */
```

Comment:
Is it needed? The factory can return only an object implementing `StateMachineInterface` if I see correctly
```suggestion
```
or if it is necessary:
```suggestion
        /** @var StateMachineInterface $stateMachine */
```